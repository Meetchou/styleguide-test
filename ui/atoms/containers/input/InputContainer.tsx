import React, { FC } from 'react';

import Label from '@ui/atoms/labels/label/Label';
import LabelSkeleton from '@ui/atoms/loaders/label-skeleton/LabelSkeleton';
import { TextFieldSkeleton } from '@ui/atoms/fields/text/TextField';
import ErrorText from '@ui/atoms/texts/error/ErrorText';

export type InputContainerProps = React.HTMLAttributes<HTMLDivElement> & {
  htmlFor?: string;
  label?: string | boolean;
  loading?: boolean;
  skeleton?: FC;
  error?: string;
};

const InputContainer:FC<InputContainerProps> = ({
  label,
  htmlFor,
  error,
  children,
  loading,
  skeleton: InputSkeleton = TextFieldSkeleton,
  ...rest
}) => {
  if (loading) {
    return (
      <div {...rest}>
        {label
          ? <LabelSkeleton />
          : null
        }
        <InputSkeleton />
      </div>
    );
  }

  return (
    <div {...rest }>
      {label
        ? (
          <Label
            htmlFor={htmlFor}
            className='mb-2'
            error={!!error}
          >
            {label}
          </Label>
        ) : null
      }
      <div className='relative'>
        {children}
      </div>
      {error
        ? <ErrorText data-testid='field-error' className='mt-1'>{error}</ErrorText>
        : null
      }
    </div>
  );
};

export default InputContainer;
