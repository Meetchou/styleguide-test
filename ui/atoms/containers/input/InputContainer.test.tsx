import React from 'react';

import render from '@ui/testing/render';

import InputContainer from './InputContainer';

describe('ui/atoms/containers/input/InputContainer', () => {
  const props = {
    label: 'test',
    htmlFor: 'field-id'
  };

  describe('renders correctly', () => {
    it('by default', () => {
      const tree = render(
        <InputContainer {...props}>
           this is a test
        </InputContainer>
      );

      expect(tree.toJSON()).toMatchSnapshot();
    });

    it('on loading', () => {
      const tree = render(
        <InputContainer {...props} loading>
           this is a test
        </InputContainer>
      );

      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
