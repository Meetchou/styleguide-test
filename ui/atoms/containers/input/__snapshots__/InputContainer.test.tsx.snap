// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/atoms/containers/input/InputContainer renders correctly by default 1`] = `
<div>
  <label
    className="block text-xs font-semibold text-gray-900 mb-2"
    htmlFor="field-id"
  >
    test
  </label>
  <div
    className="relative"
  >
    this is a test
  </div>
</div>
`;

exports[`ui/atoms/containers/input/InputContainer renders correctly on loading 1`] = `
<div>
  <div
    className="animate-pulse bg-gray-100 rounded h-3 mt-1 mb-2 w-1/2 sm:w-1/4"
  />
  <div
    className="animate-pulse bg-gray-100 rounded-lg h-10"
  />
</div>
`;
